bdsg/include/bdsg/snarl_distance_index.hpp

```diff
@@ -822,176 +999,117 @@ class SnarlDistanceIndex : public SnarlDecomposition, public TriviallySerializab
 
         //Constructors assuming that this record already exists
         SnarlTreeRecord(){};
-        SnarlTreeRecord (size_t pointer, const bdsg::yomo::UniqueMappedPointer<bdsg::MappedIntVector>* tree_records){
-            record_offset = pointer;
-            records = tree_records;
-
-#ifdef debug_distance_indexing
-            record_t type = get_record_type();
-            assert(type >= 1 && type <= 17 );
-#endif
-        }
-        SnarlTreeRecord (const net_handle_t& net, const bdsg::yomo::UniqueMappedPointer<bdsg::MappedIntVector>* tree_records){
-            record_offset = get_record_offset(net);
-            records = tree_records;
-#ifdef debug_distance_indexing
-            record_t type = get_record_type();
-            assert(type >= 1 && type <= 17 );
-#endif
-        }
+        SnarlTreeRecord (size_t pointer, const bdsg::yomo::UniqueMappedPointer<bdsg::MappedIntVector>* tree_records);
+        SnarlTreeRecord (const net_handle_t& net, const bdsg::yomo::UniqueMappedPointer<bdsg::MappedIntVector>* tree_records);
 
-        virtual size_t get_offset() { return record_offset; }
```

Comment:
Having these not be virtual is good, I think. You probably want like one field in this struct that holds the offset, and one for the pointer, so it's cheap to construct and fast to use as an accessor.